### DISCUSSION TITLE
Accessing actor properties via Dependency KeyPaths

### DISCUSSION THREAD

**Discussion by jayhickey** [pointfreeco/swift-dependencies#41] [Q&A] [answered]
Would it be possible to provide async KeyPath access to `actor` properties? I know I can `@Dependency` the entire actor and then `await` to access the property, but it would be nice to narrowly scope my dependency access to just the property I need. For example: 

```
// ❌ Cannot form key path to actor-isolated property 'myActorProperty'
@Dependency(\.myActor.myActorProperty) var myProperty
```

Even if it wasn't possible to do with dependency properties it would be useful for inline `@Dependency` access.

Thanks!

**Comment by mbrandonw** [comment, marked as answer]
Hi @jayhickey, I don't think this is possible for a number of reasons, e.g. neither key paths nor property wrappers are well integrate with async.

However, it's possible to use actors for dependencies (and encouraged) without literally making your dependency an actor. You can just use one on the inside of the live implementation. We have an example of this in our standups demo ([SpeechClient](https://github.com/pointfreeco/swiftui-navigation/blob/2f4074d2b5d02ebdb12c949ddf1a90cdf08ad1aa/Examples/Standups/Standups/Dependencies/SpeechClient.swift)) and we have a few examples over in [isowords](http://github.com/pointfreeco/isowords).

**Reply by jayhickey** [reply]
Thanks for this info @mbrandonw! I had a feeling there was probably a limitation here. Wrapping the actor sounds like a reasonable workaround, thanks for this!
